lib/datadog/profiling.rb

```diff
@@ -68,7 +68,8 @@ def self.allocation_count
 
     def self.enabled?
       profiler = Datadog.send(:components).profiler
-      !!(profiler.scheduler.running? if profiler)
+      # Use .send(...) to avoid exposing the attr_reader as an API to the outside
+      !!(profiler.send(:scheduler).running? if profiler)
```

Comment:
Should we add an instance method to the profiler that returns if it is running? 

`profiler.running?`

The method internally can call the private scheduler attribute and any other ways for it to check if is running.

That way we can get away from using `send`

Reply:
Sounds good 😄 

We can always revisit this if we ever need from another component